robot.js

```diff
@@ -94,70 +115,126 @@
+            console.log(`flag reached in ${moves} moves and ${turns} turns`);
+            if (applesEaten > 0) {
+              console.log('total apples eaten: ' + applesEaten);
+            }
+          } else if (cell === 'A') {
+            applesEaten += 1;
+            console.log('apple eaten: YUM');
+          }
+        } else {
+          console.log('move blocked by obstacle');
         }
-      } else if (cell === 'A') {
-        applesEaten += 1;
-        console.log('apple eaten: YUM');
-      }
-    } else {
-      console.log('move blocked by obstacle');
-    }
 
-    moves += 1;
-    renderBoard();
-  }
-
-  function turn(turnDirection) {
-    if (turnDirection !== 'left' && turnDirection !== 'right') {
-      console.log('ignoring invalid turn', turnDirection);
-      return;
+        moves += 1;
+        renderBoard();
     }
 
-    console.log('executing turn()');
+    function turn(turnDirection) {
+      if (turnDirection !== 'left' && turnDirection !== 'right') {
+        console.log('ignoring invalid turn', turnDirection);
+        return;
+      }
 
-    switch (robot.dir) {
-      case 'up':
-        robot.dir = turnDirection === 'left' ? 'left' : 'right';
-        break;
-      case 'down':
-        robot.dir = turnDirection === 'left' ? 'right' : 'left';
-        break;
-      case 'left':
-        robot.dir = turnDirection === 'left' ? 'down' : 'up';
-        break;
-      case 'right':
-        robot.dir = turnDirection === 'left' ? 'up' : 'down';
-        break;
-    }
+      console.log('executing turn()');
+
+      /* changing the following code into if/else statements.
+      switch (robot.dir) {
+        case 'up':
+          robot.dir = turnDirection === 'left' ? 'left' : 'right';
+          break;
+        case 'down':
+          robot.dir = turnDirection === 'left' ? 'right' : 'left';
+          break;
+        case 'left':
+          robot.dir = turnDirection === 'left' ? 'down' : 'up';
+          break;
+        case 'right':
+          robot.dir = turnDirection === 'left' ? 'up' : 'down';
+          break;
+      } */
+
+      switch (robot.dir) {
+        case 'up':
+          if (turnDirection === 'left') {
+            robot.dir = 'left';
+          } else {
+            robot.dir = 'right';
+          }
+          break;
+        case 'down':
+          if (turnDirection === 'left') {
+            robot.dir = 'right';
+          } else {
+            robot.dir = 'left';
+          }
+          break;
+        case 'left':
+          if (turnDirection === 'left') {
+            robot.dir = 'down';
+          } else {
+            robot.dir = 'up';
+          }
+          break;
+        case 'right':
+          if (turnDirection === 'left') {
+            robot.dir = 'up';
+          } else {
+            robot.dir = 'down';
+          }
+          break;
+      }
+          turns += 1;
+          renderBoard();
+      }
 
-    turns += 1;
-    renderBoard();
-  }
+      renderAll();
+    }) ();
```

Comment:
This line should start in column 1 (just do Format Document before you save a file).